src/rpc/blockchain.cpp

```diff
@@ -78,20 +78,22 @@ double GetDifficulty(const CChain& chain, const CBlockIndex* blockindex)
     return dDiff;
 }
 
-double GetDifficulty(const CBlockIndex* blockindex)
+static int ComputeBlockConfirmations(const CBlockIndex* tip, const CBlockIndex* blockindex, const CBlockIndex*& next)
 {
-    return GetDifficulty(chainActive, blockindex);
+    next = tip->GetAncestor(blockindex->nHeight + 1);
+    if (next && next->pprev == blockindex) {
```

Comment:
Why? You can just do a GetAncestor for blockindex->nHeight here, no?

Reply:
This way you get next of blockindex.

Reply:
Ohoh, sorry, missed its use in blockheaderToJSON.

Reply:
This confused me for a minute too. Maybe could rename function to something like `ComputeNextBlockAndDepth` to mention the `next` part.

Reply:
Can do, WDYT about returning `std::pair<int, const CBlockIndex*>`?

Reply:
I can add a comment there too explaning the `height + 1` and `->pprev == blockindex`.

Reply:
Renamed to `ComputeNextBlockAndDepth` as per @ryanofsky suggestion.